src/base-input.js

```diff
@@ -550,6 +553,14 @@ export default class BaseInput extends LitElement {
     if (this.hasBlurred && this.type === 'credit-card') {
       this.validate();
     }
+
+    // Enables keyup validation
+    const regex = new RegExp("^[a-zA-Z0-9]+$", "u");
+    const key = String.fromCharCode(!evt.charCode ? evt.which : evt.charCode);
```

Comment:
I don't understand why we need this regex. Why not respond to all keys? This excludes characters like `'` or `#` which could also be typed into the input.

Reply:
I'm confused by `+$` which "matches the end of the line". When I run this through a regex tester it stops matching all the regular test content.

Also, I agree with Jeff. The a-zA-Z makes the regex exclude certain characters that should probably be in there (such as `ń` and `ñ`)/

Reply:
I thought about this over the weekend. The reason for the regex is to not validate on keys like `shift` and `delete`, but the better fix is not to bind this to the `keyup` event, but the `input` event.